packages/core/src/providers/user/connectWalletProvider/index.tsx

```diff
@@ -302,6 +313,173 @@ export const ConnectWalletProvider = ({
     () => !!mainProxiesAccounts?.length,
     [mainProxiesAccounts?.length]
   );
+  const {
+    mutateAsync: onBackupGoogleDrive,
+    isLoading: backupGoogleDriveLoading,
+    isSuccess: backupGoogleDriveSuccess,
+  } = useMutation({
+    mutationFn: async ({
+      account: tradeAccount,
+      password,
+    }: ExportTradeAccountProps) => {
+      tradeAccount.isLocked && tradeAccount.unlock(password);
+      const jsonAccount = tradeAccount.toJson(password);
+      if (!gDriveReady) {
+        await GoogleDrive.init();
+        setGDriveReady(true);
+      }
+      await GoogleDrive.addFromJson(jsonAccount);
+      await sleep(2000);
+      await onRefetchGoogleDriveAccounts();
+    },
+    onError: (error: { message: string }) =>
+      onHandleError(error?.message ?? error),
+  });
+
+  const hasLocalToken = useMemo(
+    () => localStorageOrDefault(GOOGLE_LOCAL_STORAGE_KEY, null, true),
+    []
+  );
+
+  const {
+    refetch: onConnectGoogleDrive,
+    isLoading: connectGoogleDriveLoading,
+    isFetching: connectGoogleDriveFetching,
+    isSuccess: connectGoogleDriveSuccess,
+  } = useQuery({
+    staleTime: 100000,
+    enabled: !!hasLocalToken && googleDriveStore,
+    refetchOnMount: false,
+    refetchOnReconnect: false,
+    refetchOnWindowFocus: false,
+    retry: false,
+    initialData: undefined,
+    queryKey: [QUERY_KEYS.googleSession()],
+    queryFn: async () => {
+      if (!gDriveReady) {
+        await GoogleDrive.init();
+        setGDriveReady(true);
+      }
+      return true;
+    },
+  });
+
+  const { data: googleDriveAccounts, refetch: onRefetchGoogleDriveAccounts } =
+    useQuery({
+      queryKey: QUERY_KEYS.googleAccounts(),
+      enabled: gDriveReady,
+      initialData: [],
+      queryFn: async () => {
+        const accounts = await GoogleDrive.getAll();
+        return accounts.map((account) => keyring.createFromJson(account));
+      },
+      onError: (error: { message: string }) => {
+        onHandleError(error?.message ?? error);
+      },
+    });
+
+  const onAddAccountFromJson = useCallback(
+    (json: KeyringPair$Json) => GoogleDrive.addFromJson(json),
+    [GoogleDrive]
+  );
+
+  const {
+    error: registerError,
+    mutateAsync: onRegisterTradeAccount,
+    status: registerStatus,
+  } = useAddProxyAccount({
+    onError: (e: Error) => onHandleError(e.message),
+    onSuccess: (msg) => msg && onHandleAlert(msg),
+    onSetTempMnemonic,
+    onRefetchGoogleDriveAccounts,
+    onAddAccountFromJson,
+  });
+  const {
+    mutateAsync: onRemoveGoogleDrive,
+    isLoading: removeGoogleDriveLoading,
+    isSuccess: removeGoogleDriveSuccess,
+  } = useMutation({
```

Comment:
I think `useMutation` and `useQuery` shouldn't be called here directly. We should rather create a new folder `core/hooks/gDrive` and put all gDrive related stuff there. Please define all related hooks there.

Reply:
You can import hooks from there and use them here.